src/software/ai/evaluation/detect_successful_kick_test.cpp

```diff
@@ -0,0 +1,44 @@
+#include "software/ai/evaluation/detect_successful_kick.h"
+
+#include <gtest/gtest.h>
+
+
+TEST(DetectSuccessfulKickTest, ball_over_speed_threshold_and_no_direction_difference)
+{
+    Ball ball({0, 0}, {5, 5}, Timestamp::fromSeconds(0));
+
+    Vector expected_direction(4, 4);
+
+    EXPECT_TRUE(successfulKickDetected(ball, expected_direction));
+}
+
+TEST(DetectSuccessfulKickTest, ball_under_speed_threshold_and_no_direction_difference)
+{
+    Ball ball({5, 2}, {0, 0.2}, Timestamp::fromSeconds(0));
+
+    Angle angle               = angle.fromDegrees(90);
+    Vector expected_direction = Vector::createFromAngle(angle);
+
+    EXPECT_FALSE(successfulKickDetected(ball, expected_direction));
+}
+
+TEST(DetectSuccessfulKickTest,
+     ball_over_speed_threshold_and_small_negative_direction_difference)
```

Comment:
Can we have a test for small positive direction difference too?

Reply:
added test